core/lib/dal/src/blocks_dal.rs

```diff
@@ -190,6 +192,34 @@ impl BlocksDal<'_, '_> {
         Ok(row.map(|row| L1BatchNumber(row.number as u32)))
     }
 
+    /// Gets a number of the last L1 batch that is ready for commitment generation (i.e., doesn't have commitment
+    /// yet, and has tree data).
+    pub async fn get_last_l1_batch_ready_for_commitment_generation(
```

Comment:
Does a separate query make sense, or would it make sense to extend `get_next_l1_batch_ready_for_commitment_generation()` to return `ops::RangeInclusive` and use a single DB query?

Reply:
IMO, separate query makes sense